### DISCUSSION TITLE
Why should a page `pathLocale` be customizable

### DISCUSSION THREAD

**Discussion by Mister-Hope** [DreamOfIce/vuepress-plugin-i18n#12] [Q&A] [answered]
VuePress internally mark pages with `locale` object, and this is a core feature of i18n support. Any page inside a locale folder will be isolated with that locale.

Why should an user set `_i18n.pathLocale` in frontmatter to override the locale a page belongs to, e.g.: `/zh/example.html` to `/en/` ?

**Comment by DreamOfIce** [comment]
Frontmatter `_i18n` is just an internal field and normally users do not need to know this.
It used only when call `createPage()` inside the plugin to pass some data to `extendsPage` and will be deleted  at that time:

https://github.com/DreamOfIce/vuepress-plugin-i18n/blob/fc9cfd908dfe4cd25c3feadaa29d7c4254654c9a/src/node/lib/untranslated.ts#L28-L44

https://github.com/DreamOfIce/vuepress-plugin-i18n/blob/fc9cfd908dfe4cd25c3feadaa29d7c4254654c9a/src/node/lib/pageData.ts#L15-L24

**Reply by Mister-Hope** [reply]
Every page should have `pathLocale`, so the question is still there, why there is a `frontmatter._i18n.pathLocale`?

**Reply by Mister-Hope** [reply]
If users do not need to set this field, then we absolutely do not need this field. I think the thing you should explain is that why users should sometimes set this field?

**Reply by DreamOfIce** [reply, marked as answer]
The root cause is: for auto-filled (I can't find a better expr at the moment) untranslated pages, we can only set the filePathRelative in the `extendsPage` hook.
This also causes the pathLocale property to always be `/`, so we have to override it manually

**Reply by DreamOfIce** [reply]
Regarding why the filePathRelative must be written in `extendsPage`:
Because theme-hope will only add pages with a non-empty `filePathRelative` attribute to the sidebar using "structure" mode, but the non-existent `filePathRelative` will make the git plugin throw an error.

This is the only way I can think of at the moment, but if you have a better solution that would be great!

**Reply by Mister-Hope** [reply]
I think I got you, so when you `createPage` to add a page, the page will not get a correct `pathLocale` right?

**Reply by DreamOfIce** [reply]
Yes, when `filePathRelative` is empty, `pathLocale` is always `/`:
https://github.com/vuepress/vuepress-next/blob/d91996fd0864abd5029e4c4cff319db48be06b47/packages/core/src/page/inferPagePath.ts#L17-L22
